photon-ml/src/main/scala/com/linkedin/photon/ml/function/svm/SmoothedHingeLossLinearSVMFunction.scala

```diff
@@ -12,33 +12,19 @@
  * License for the specific language governing permissions and limitations
  * under the License.
  */
-package com.linkedin.photon.ml.function
+package com.linkedin.photon.ml.function.svm
 
-import breeze.linalg
+import breeze.linalg.Vector
 
 import com.linkedin.photon.ml.constants.MathConst
 import com.linkedin.photon.ml.data.LabeledPoint
 
 /**
  * Implement Rennie's smoothed hinge loss function (http://qwone.com/~jason/writing/smoothHinge.pdf) as an
- * optimizer-friendly approximation for linear SVMs
+ * optimizer-friendly approximation for linear SVMs. This Object is to the individual/distributed smoothed hinge loss
+ * functions is as the PointwiseLossFunction is to the individual/distributed GLM loss functions.
  */
-class SmoothedHingeLossFunction extends DiffFunction[LabeledPoint] {
-  override protected[ml] def calculateAt(
-      datum: LabeledPoint,
-      coefficients: linalg.Vector[Double],
-      cumGradient: linalg.Vector[Double]): Double = {
-
-    val margin = datum.computeMargin(coefficients)
-    val (loss, deriv) = SmoothedHingeLossFunction.lossAndDerivative(margin, datum.label)
-
-    // Eq. 5, page 2 (derivative multiplied by label in lossAndDerivative method)
-    breeze.linalg.axpy(datum.weight * deriv, datum.features, cumGradient)
-    datum.weight * loss
-  }
-}
-
-object SmoothedHingeLossFunction {
+object SmoothedHingeLossLinearSVMFunction {
```

Comment:
I'm not sure if renaming this is quite right. A smooth hinge loss function is actually distinct from SVM, though probably its most common application is in SVM classifiers.